idea/resources/META-INF/jvm-common.xml

```diff
@@ -137,6 +137,8 @@
             displayName="Incompatible API usage"
             implementationClass="org.jetbrains.kotlin.idea.inspections.api.IncompatibleAPIInspection"/>
 
+        <projectService serviceInterface="org.jetbrains.uast.kotlin.BaseKotlinUastResolveProviderService"
+                        serviceImplementation="org.jetbrains.uast.kotlin.internal.IdeaKotlinUastResolveProviderService"/>
         <projectService serviceInterface="org.jetbrains.uast.kotlin.KotlinUastResolveProviderService"
                         serviceImplementation="org.jetbrains.uast.kotlin.internal.IdeaKotlinUastResolveProviderService"/>
```

Comment:
Why do we registering IdeaKotlinUastResolveProviderService two times?

Reply:
I explained this in the PR description:

> One downside is, service registration is using an interface _name_ as a key, so we need to register the same instance of plugin-specific service with two different keys: one is "base" service interface, and the other is plugin-specific service interface where APIs could diverge, e.g., retrieval of `BindingContext`.

One option I can think of is to consolidate all APIs, even including frontend-specific APIs, to the base service, and then to override those APIs in the counterpart plugin as an error. E.g., retrieval of `BindingContext` is regarded as an error in FIR UAST service. This way, we can register only one service instance. I added this change as a separate commit (5th), just in case.

Reply:
It is not a good to have some fe10 parts in base interface. We may get the dependencies on fe10 compiler from fir ide in the future, so it is better to keep them separate. Let's get back to the option with two registrations of the same services (without 5ths commit)

Reply:
Glad I made it a separate commit so that I can easily drop that. :D Reverted.